Bottom: bb02bd99c67f1a2740ffbf00c8d1c4bfffbceb37
Top:    13b86d0e7b9b6f9c44b40d9b12c409a8d77545d1
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-07 02:18:22 +0000

polygon1.c: Fix Collect routine

Not 100% sure about this, but it seems to fix a nasty
case regarding incorrect geometry created in the following
test geometry:

Element["" "" "" "unknown" 1325.00mil 1070.00mil -175.00mil -105.00mil 0 100 ""]
(
  Pad[-85.00mil -39.00mil -79.00mil -39.00mil 34.00mil 30.00mil 40.00mil "2" "2" "square"]
)

Layer(1 "component")
(
  Line[1245.00mil 1030.00mil 1195.00mil 1030.00mil 12.00mil 20.00mil "clearline"]
  Polygon("clearpoly")
  (
    [5.00mil 5.00mil] [4195.00mil 5.00mil] [4195.00mil 2695.00mil] [5.00mil 2695.00mil]
  )
)


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 95ce14f..f721685 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1664,6 +1664,10 @@ Collect (char poly, jmp_buf * e, PLINE * a, POLYAREA ** contours, PLINE ** holes
 
       if (j_rule (poly, cur, &dir) && cur->Flags.mark == 0)
 	Collect1 (e, cur, dir, contours, holes, j_rule);
+//    The lines below are no-longer required.. preserving for now to record this commit.
+//      other = cur;
+//      if (other->cvc_prev)
+//	Collect1 (e, other, dir, contours, holes, j_rule);
     }
   while ((cur = cur->next) != &a->head);
 }				/* Collect */
